### PR TITLE
Registry Trait: fix strtoupper error

### PR DESCRIPTION
**Pull request by Eren5960**
Fixed incorrect strtoupper of incorrect letters on operating systems using languages other than English. 
Like (Stair => STAİR)

## Introduction
```
Error: No such registry member: pocketmine\block\utils\StairShape::STRAIGHT
File: src/utils/RegistryTrait
Line: 102
Type: Error
```

### CODE REVIEW

**Comment by dktapps** [comment]
Please add unit tests to demonstrate the problem and to show that your change addresses it.

**Comment by Eren5960** [comment]
Test #1 with English locale:
DeepinScreenshot_select-area_20200613105256

Test #1 with Turkish locale:
DeepinScreenshot_select-area_20200613105332

**Comment by SOF3** [comment]
Please commit the unit test files not the screenshots.

**Comment by Eren5960** [comment]
I don't know which directory to upload

**Comment by dktapps** [comment]
I can't reproduce this on Windows.

**Comment by dktapps** [comment]
I've written unit tests for this bug and got it to reproduce on Travis:  https://travis-ci.com/github/pmmp/PocketMine-MP/builds/171222517#L1521

It seems like this won't be the only place bugs like this show up; for example, it's likely they'll appear  if you for some reason use `@priority highest` instead of `@priority HIGHEST` when the system language is set to Turkish. It may also happen when using `ItemFactory::fromString()`, `Enchantment::fromString()` and various other places.

TL;DR: PHP sucks.
